### PR TITLE
example-app: Move to examples/instrumentation; Added Native Histograms.

### PR DESCRIPTION
**Pull request by bwplotka**
This is done for 3 reasons:

* Enables adding more examples - we already have planned client-server one ("pinger")
* To separate core components with example applications
* example-app naming was confusing. It is used for both this app as well as examples/example-app.yaml referenced by the public docs. Let's be specific now.

I plan to also add more synthetic metrics and tests later on.

### CODE REVIEW

**Comment by bwplotka** [comment]
This will hopefully work seamlessly with our CI image pipeline, just instead of `gcr.io/gke-release/prometheus-engine/example-app:<tag>` to `gcr.io/gke-release/prometheus-engine/go-synthetic:<tag>`

Will fix internal playbooks later on to accommodate this.

**Review by TheSpiritXIII — APPROVED** [review]
Great idea!

File: examples/instrumentation/go-synthetic/main.go [hunk 1]
```diff
@@ -30,36 +30,44 @@ import (
...
+	maxCount             = flag.Int("max-count", labelCombinations, "Maximum metric instance count for all metric types.")
+	histogramCount       = flag.Int("histogram-count", 2, "Number of unique instances per histogram metric.")
+	nativeHistogramCount = flag.Int("native-histogram-count", -1, "Number of unique instances per native-histogram metric."+
+		"Note that native histograms are not supported in text format exposition, so traditional protobuf format has to be enabled on your collector. See https://prometheus.io/docs/prometheus/latest/feature_flags/#native-histograms")
+	gaugeCount   = flag.Int("gauge-count", -1, "Number of unique instances per gauge metric.")
+	counterCount = flag.Int("counter-count", -1, "Number of unique instances per counter metric.")
+	summaryCount = flag.Int("summary-count", -1, "Number of unique instances per summary metric.")
+
+	omStateSetCount       = flag.Int("om-stateset-count", -1, "Number of OpenMetrics StateSet metrics (https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#stateset). Requires OpenMetrics format to be negotiated.")
```
Comment (bwplotka) [inline examples/instrumentation/go-synthetic/main.go:49]: Bit misleading as those as om* flags are not yet implemented (will be in the next PR).

**Review by pintohutch — COMMENTED** [review]
LGTM!

**Pull request merged**